test/component/uk/gov/hmrc/individualincomedesstub/EmploymentSpec.scala

```diff
@@ -0,0 +1,130 @@
+      val response = requestCreateEmployment(request)
+
+      Then("The response code should be 201 (Created)")
+      response.code shouldBe 201
+
+      And("The employment is returned in the response body")
+      Json.parse(response.body) shouldBe Json.parse(createResponse)
+
+      And("The employment is stored in mongo")
+      val employment = Json.parse(response.body).as[Employment]
+      val storedEmployment = result(employmentRepository.findByReferenceAndNino(employerReference, Nino(validNino)), timeout)
+      storedEmployment shouldBe Seq(employment)
+    }
+
+    scenario("Request fails for invalid NINO") {
+      Given("A valid create employment request")
+      val request = createRequest
+
+      When("I request to create an employment for an invalid NINO")
+      val result = requestCreateEmployment(request, nino = "A12345C")
+
+      Then("The response code should be 400 (Bad Request)")
+      result.code shouldBe 400
+      result.body shouldBe """{"code":"INVALID_REQUEST","message":"Malformed nino submitted"}"""
+    }
+
+    scenario("Request fails for unencoded employer reference") {
+      Given("A valid create employment request")
+      val request = createRequest
+
+      When("I request to create an employment for an unencoded employer reference")
+      val result = requestCreateEmployment(request, reference = employerReference)
+
+      Then("The response code should be 404 (Not Found)")
+      result.code shouldBe 404
+    }
+  }
+
+  private def createResponse = {
+    s"""
+       |{
+       |"employerPayeReference":"$employerReference",
+       |"nino":"$validNino",
+       |"startDate": "2016-01-01",
+       |"endDate": "2017-03-01",
+       |"payments": [
+       |    {
+       |        "paymentDate": "2016-01-28",
+       |        "taxablePayment": 1000.55,
+       |        "nonTaxablePayment": 0
+       |    },
+       |    {
+       |        "paymentDate": "2016-02-28",
+       |        "taxablePayment": 950.55,
+       |        "nonTaxablePayment": 0
+       |    }
+       |]
+       |}""".stripMargin.replaceAll("\n", "")
+  }
+
+  private def createRequest = {
+    s"""
+       |{
+       |"startDate": "2016-01-01",
+       |"endDate": "2017-03-01",
+       |"payments": [
+       |    {
+       |        "paymentDate": "2016-01-28",
+       |        "taxablePayment": 1000.55,
+       |        "nonTaxablePayment": 0
+       |    },
+       |    {
+       |        "paymentDate": "2016-02-28",
+       |        "taxablePayment": 950.55,
+       |        "nonTaxablePayment": 0
+       |    }
+       |]
+       |}
+     """.stripMargin.replaceAll("\n", "")
+  }
+
+  private def requestCreateEmployment(request: String, reference: String = employerReferenceEncoded, nino: String = validNino) = {
+    Http(s"$serviceUrl/employer/$reference/employment/$nino")
+      .postData(Json.parse(request).toString())
+      .headers(HeaderNames.CONTENT_TYPE -> "application/json"
```

Comment:
Just as there's a constant for `HeaderNames.CONTENT_TYPE` there's also one for the mime types; `play.api.http.MimeTypes.JSON`.